docs/resources/network_policy.md

```diff
@@ -0,0 +1,75 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "materialize_network_policy Resource - terraform-provider-materialize"
+subcategory: ""
+description: |-
+  A network policy manages access to the system through IP-based rules.
+---
+
+# materialize_network_policy (Resource)
+
+A network policy manages access to the system through IP-based rules.
+
+## Example Usage
+
+```terraform
+resource "materialize_network_policy" "office_policy" {
+  name = "office_access_policy"
+
+  rule {
+    name      = "minnesota"
+    action    = "allow"
+    direction = "ingress"
+    address   = "2.3.4.5/32"
+  }
+
+  rule {
+    name      = "new_york"
+    action    = "allow"
+    direction = "ingress"
+    address   = "1.2.3.4/28"
+  }
+
+  comment = "Network policy for office locations"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The name of the network policy.
+- `rule` (Block Set, Min: 1, Max: 25) Rules for the network policy. (see [below for nested schema](#nestedblock--rule))
+
+### Optional
+
+- `comment` (String) **Public Preview** Comment on an object in the database.
+- `region` (String) The region to use for the resource connection. If not set, the default region is used.
```

Comment:
I'm unfamiliar with the Terraform provider, is `region` a field we typically allow folks to specify?

Reply:
Yep, this was introduced last year when we refactored the provider, this allows users to manage cross region instances within the same terraform project.

For example:

```hcl
resource "materialize_database" "database" {
  name    = "example_database"
  comment = "database comment"
}

# Create in separate region
resource "materialize_database" "database_us_west" {
  name    = "example_database"
  comment = "database comment"
  region  = "aws/us-west-2"
}
```

the first one will create the db in the default region defined in the provider configuration, but the second one overrides the region and will create the db in `aws/us-west-2` if it is enabled.